webrender/src/device/gfx/descriptor.rs

```diff
@@ -132,9 +133,21 @@ impl<B: hal::Backend> DescPool<B> {
 }
 
 #[derive(Copy, Clone, Debug)]
-pub(super) struct DescriptorSetLocation {
+pub(super) struct DescriptorSetInfo {
     pool_idx: usize,
     set_idx: usize,
+    pub bound_in_frame: GpuFrameId,
+}
+
+impl DescriptorSetInfo {
+    pub fn still_in_flight(&self, frame_id: GpuFrameId, frame_count: usize) -> bool {
+        for i in 0 .. frame_count {
+            if self.bound_in_frame == GpuFrameId(frame_id.0 - i) {
```

Comment:
we should make this a single comparison, i.e. `self.bound_in_frame.0 > frame_id.0 - frame_count`